jetbrains-core/src/software/aws/toolkits/jetbrains/services/federation/psireferences/ArnPsiReferenceContributor.kt

```diff
@@ -0,0 +1,34 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.federation.psireferences
+
+import com.intellij.patterns.PsiElementPattern
+import com.intellij.psi.ElementManipulators
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiReferenceContributor
+import com.intellij.psi.PsiReferenceRegistrar
+import com.intellij.util.ProcessingContext
+
+class ArnPsiReferenceContributor : PsiReferenceContributor() {
+    override fun registerReferenceProviders(registrar: PsiReferenceRegistrar) {
+        registrar.registerReferenceProvider(
+            object : PsiElementPattern.Capture<PsiElement>(
+                PsiElement::class.java
+            ) {
+                override fun accepts(o: Any?, context: ProcessingContext): Boolean {
+                    if (o == null || o !is PsiElement) return false
+                    val manipulator = ElementManipulators.getManipulator(o)
+                    if (manipulator == null) return false
+
+                    if (manipulator.getRangeInElement(o).substring(o.text).contains("arn:")) {
```

Comment:
Should this be assigned to a variable as `arn_prefix`

Reply:
if we start using this everywhere sure, but this is one of 2 places at the moment